### PR TITLE
use apps/v1 instead of extensions/v1beta1 for deployment

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
@jianzhangbjz I think that `Service Catalog should check basic usages: OCP-24062, OCP-24049, OCP-15600` is unstable for example compare [failed](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24305/pull-ci-openshift-origin-master-e2e-gcp/4650) with [passed](https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24305/pull-ci-openshift-origin-master-e2e-gcp/4646). It would be nice it could be improved. 

You can find more traces in 
https://prow.svc.ci.openshift.org/job-history/origin-ci-test/pr-logs/directory/pull-ci-openshift-origin-master-e2e-gcp

**Comment by p0lyn0mial** [comment]
again it passed https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24305/pull-ci-openshift-origin-master-e2e-gcp/4660

**Comment by p0lyn0mial** [comment]
again it passed https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24305/pull-ci-openshift-origin-master-e2e-gcp/4673

File: test/extended/testdata/service_catalog/ups-deployment.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: Template
 metadata:
   name: ups-deployment-template
 objects:
-- apiVersion: extensions/v1beta1
+- apiVersion: apps/v1
```
Comment (tnozicka) [inline test/extended/testdata/service_catalog/ups-deployment.yaml]: /hold
doesn't this need updating the bindata? and why isn't verify screaming?

Comment (p0lyn0mial) [reply test/extended/testdata/service_catalog/ups-deployment.yaml]: it does, filling a BZ

**Comment by tnozicka** [comment]
bindata updated
/hold cancel
/lgtm

@p0lyn0mial is filling a BZ for the broken verify

File: test/extended/testdata/service_catalog/ups-deployment.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply test/extended/testdata/service_catalog/ups-deployment.yaml]: https://bugzilla.redhat.com/show_bug.cgi?id=1784324, thanks.

**Comment by mfojtik** [comment]
/override ci/prow/e2e-cmd

This test is unrelated to the change which is holding the 1.17 rebase.

**Pull request merged**